src/test/java/reposense/authorship/AnnotatorAnalyzerTest.java

```diff
@@ -0,0 +1,69 @@
+package reposense.authorship;
+
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import reposense.authorship.model.FileInfo;
+import reposense.authorship.model.FileResult;
+import reposense.authorship.model.LineInfo;
+import reposense.model.Author;
+import reposense.parser.InvalidLocationException;
+import reposense.template.GitTestTemplate;
+import reposense.util.TestUtil;
+
+public class AnnotatorAnalyzerTest extends GitTestTemplate {
+
+    private static final Author MAIN_AUTHOR = new Author(MAIN_AUTHOR_NAME);
+    private static final Author FAKE_AUTHOR = new Author(FAKE_AUTHOR_NAME);
+    private static final Date SINCE_DATE = TestUtil.getSinceDate(2018, Calendar.FEBRUARY, 8);
+    private static final Date UNTIL_DATE = TestUtil.getUntilDate(2018, Calendar.FEBRUARY, 10);
+
+    @Before
+    public void before() throws InvalidLocationException {
+        super.before();
+        config.setSinceDate(SINCE_DATE);
+        config.setUntilDate(UNTIL_DATE);
+    }
+
+    @Test
+    public void applyAnnotationTest() {
+        config.setAuthorList(Arrays.asList(FAKE_AUTHOR, MAIN_AUTHOR));
+        FileResult fileResult = getFileResult("annotationTest.java");
+        assertAnnotationAnalysisCorrectness(fileResult, FAKE_AUTHOR, MAIN_AUTHOR);
+    }
+
+    @Test
+    public void disownCodeTest() {
+        config.setAuthorList(Collections.singletonList(FAKE_AUTHOR));
+        FileResult fileResult = getFileResult("annotationTest.java");
+        assertAnnotationAnalysisCorrectness(fileResult, FAKE_AUTHOR, Author.UNKNOWN_AUTHOR);
+    }
+
+    public FileResult getFileResult(String relativePath) {
+        FileInfo fileInfo = FileInfoExtractor.generateFileInfo(config.getRepoRoot(), relativePath);
+        return FileInfoAnalyzer.analyzeFile(config, fileInfo);
+    }
+
+    /**
+     * Asserts the correctness of authorship overriding with regards to @@author tags given
+     * in the test file.
+     */
+    public void assertAnnotationAnalysisCorrectness(FileResult fileResult, Author originalAuthor,
+            Author overridingAuthor) {
+        for (LineInfo line : fileResult.getLines()) {
+            Author lineAuthor = line.getAuthor();
+            if (line.getContent().startsWith("fake")) {
```

Comment:
perhaps the test methods need a little rework; this would limit the usage of this method (for future test cases) isnt it?

Reply:
I've modified the test method. The test method now accepts the actual `FileResult` obtained  from analyzing the test file, and a `List<Author>` that contains the expected author for each line in the test file.

Reply:
@eugenepeh  @fzdy1914  may I get your review on the changes?